c/tskit/core.h

```diff
@@ -67,6 +69,33 @@ extern "C" {
 #define TSK_DBL_DECIMAL_DIG (DBL_DIG + 3)
 #endif
 
+/* We define a specific NAN value for default mutation time which indicates
+ * the time is unknown. We use a specific value so that if mutation time is set to
+ * a NAN from a computation we can reject it.
+ */
+#define TSK_MUTATION_UNKNOWN_TIME_HEX 0x7FF874736B697421ULL
+static inline double
+__tsk_nan_f(void)
+{
+    const union {
+        uint64_t i;
+        double f;
+    } nan_union = { TSK_MUTATION_UNKNOWN_TIME_HEX };
+    return nan_union.f;
+}
+static inline bool
+__tsk_is_nan_f(double val)
+{
+    union {
+        uint64_t i;
+        double f;
+    } nan_union;
+    nan_union.f = val;
+    return nan_union.i == TSK_MUTATION_UNKNOWN_TIME_HEX;
+}
+#define TSK_MUTATION_UNKNOWN_TIME __tsk_nan_f()
+#define TSK_IS_MUTATION_UNKNOWN_TIME(val) __tsk_is_nan_f(val)
```

Comment:
Should we just make this a function altogether? ``is_mutation_unknown_time(val)`` seems a bit more natural. Or, to put it another way, is there any particular reason for making this a function-like macro rather than just function?

Reply:
I think I wanted to keep it all together - but yes no need to be a macro. Fixed in https://github.com/tskit-dev/tskit/pull/672/commits/9b742c8bc1e5fb298dff844af0b56a9975ea605f